models/repo_collaboration.go

```diff
@@ -186,9 +186,46 @@ func (repo *Repository) DeleteCollaboration(uid int64) (err error) {
 		return err
 	}
 
+	// Unassign a user from any issue (s)he has been assigned to in the
+	// repository
+	if err := repo.removeIssueAssignees(sess, uid); err != nil {
+		return err
+	}
+
 	return sess.Commit()
 }
 
+func (repo *Repository) removeIssueAssignees(e Engine, userID int64) error {
+
+	err := repo.GetOwner()
+	if err != nil {
+		// Very weird case
+		return fmt.Errorf("removeIssueAssignees: Unable to load owner for repo: %d Error: %v", repo.ID, err)
+	}
+
+	perm, err := getUserRepoPermission(e, repo, repo.Owner)
+	if err != nil {
+		return err
+	}
+
+	if perm.CanWrite(UnitTypeIssues) {
```

Comment:
Careful! It should be `UnitTypePullRequests` in the case of PRs.

However, I think you should use `models.CanBeAssigned()` in an issue-by-issue basis instead; that would cover all cases and will always be consistent.

Reply:
I have updated it to make use of `CanBeAssigned`